src/app/server/Server.h

```diff
@@ -309,6 +313,23 @@ struct CommonCaseDeviceServerInitParams : public ServerInitParams
         }
 #endif
 
+#if CHIP_CONFIG_TC_REQUIRED
+        static app::DefaultTermsAndConditionsProvider sDefaultTermsAndConditionsProviderInstance;
+        static app::DefaultTermsAndConditionsStorageDelegate sDefaultTermsAndConditionsStorageDelegateInstance;
+
+        Optional<app::TermsAndConditions> termsAndConditions = Optional<app::TermsAndConditions>((app::TermsAndConditions){
+            .value   = CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS,
+            .version = CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION,
+        });
+
+        ReturnErrorOnFailure(sDefaultTermsAndConditionsStorageDelegateInstance.Init(this->persistentStorageDelegate));
+        ReturnErrorOnFailure(sDefaultTermsAndConditionsProviderInstance.Init(&sDefaultTermsAndConditionsStorageDelegateInstance,
+                                                                             termsAndConditions));
+        this->termsAndConditionsProvider = &sDefaultTermsAndConditionsProviderInstance;
```

Comment:
This means that in this case (which most products will inherit blindly) will use the CHIP_CONFIG_XXX per-firmware version. Would recommend not doing any magic init of this here, and making every user of T&C provider setup explicit init of the T&C when needed.

Reply:
I moved the initialization logic to AppTask.cpp. Let me know if it's okay.